api/pkg/apis/v1alpha1/providers/target/helm/helm.go

```diff
@@ -723,6 +742,16 @@ func configureInstallClient(ctx context.Context, name string, componentProps *He
 	// This should added when we upgrade to helm ^3.13.1
 	return installClient, nil
 }
+func checkReleaseExists(config *action.Configuration, releaseName string) bool {
+	historyClient := action.NewHistory(config)
+	historyClient.Max = 1
+	if _, err := historyClient.Run(releaseName); err != nil {
+		if strings.Contains(err.Error(), "release: not found") {
```

Comment:
It is not a good way to use string compare to determine error type, the error string is returned by helm tool, it may change in the future. We'd better to follow the pattern like this, 
https://github.com/dokku/dokku/blob/095995da8770d705a36652dddb09fb957d4f4ff4/plugins/scheduler-k3s/helm.go#L157C1-L178C1

And I also think it is better we return bool, error instead of bool only. We can also log the error type in caller.

Reply:
Thanks, I will follow this pattern.

Reply:
**upgrade.Install = true should be used to implement the behavior like helm upgrade --install** Install failed twice will also throw the wrong error(can not re-use a name)